lib/smack/SmackRep.cpp

```diff
@@ -239,16 +340,16 @@ const Expr* SmackRep::lit(const llvm::Value* v) {
       Expr::lit(exponentPart));
 
   } else if (llvm::isa<llvm::ConstantPointerNull>(v))
-    return Expr::lit(0, width);
+    return Expr::id("$NULL");
 
   else
     return expr(v);
   // assert( false && "value type not supported" );
 }
 
-const Expr* SmackRep::lit(unsigned v) {
-  // TODO why doesn't this one do the thing with negative as well?
-  return Expr::lit(v, width);
+// Shaobo: if we add multiple types to SMACK, then integer literals generated by SMACK source code should have the sense of type widths
+const Expr* SmackRep::lit(int v, unsigned size) {
```

Comment:
This method should be removed, and calls to it replaced by others with clearer semantics regarding how the integer constant is used. For example, when this method is called to generate an integer used as `size_t`, the call should be replaced with `SmackRep::size(unsigned n)` which will generate a Boogie expression of the appropriate type, e.g., either an `int`, or a `bvXX`, where `XX` is the target-specific width of `size_t`.